config/machine-api-operator-deployment.yaml

```diff
@@ -0,0 +1,85 @@
+---
+apiVersion: apps/v1beta2
+kind: Deployment
+metadata:
+  name: machine-api-operator
+  namespace: openshift-machine-api
```

Comment:
can you elaborate why we need this file?

Reply:
This is required for the k8s-e2e ci job which runs test against the k8s. In the original deployment, we are mounting a secret into the pod. Based on the annotation in the mao metrics service object, this [`secret`](https://github.com/openshift/machine-api-operator/pull/368/files/5d2b38bc7a0e838f2032a29c6efb7c966c256912#diff-273b08d16d3abfbf3bc5df0438e0c450R91) will get created automatically in the openshift cluster.  With the minikube cluster, this secret will not be there and thus deployment will fail. To workaround this, in the k8s-e2e tests this deployment will be used which is not mounting secrets.